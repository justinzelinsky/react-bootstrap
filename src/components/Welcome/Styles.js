import BootstrapButton from 'react-bootstrap/Button';
import BootstrapJumbotron from 'react-bootstrap/Jumbotron';
import styled from 'styled-components';
import colors from 'styles/colors';
import devices from 'styles/devices';

const Button = styled(BootstrapButton)`
  margin: 30px auto 0 auto;
  width: 300px;
  display: block;
`;

const Jumbotron = styled(BootstrapJumbotron)`
  @media ${devices.phone} {
    position: fixed;
    top: 50%;
    transform: translate(0, -50%);
  }

  @media ${devices.desktop} {
    left: 50%;
    position: fixed;
    top: 50%;
    transform: translate(-50%, -50%);
  }
`;

const Header = styled.h1`

`;

const Logo = styled.img`
  width: 150px;

  @media ${devices.phone} {
    width: 75px;
  }
`;

const GetStarted = styled.div`
  font-size: 18px;
  text-align: center;

  code {
    color: ${colors.blue}
  }
`;

export default {
  Button,
  GetStarted,
  Header,
  Jumbotron,
  Logo
};