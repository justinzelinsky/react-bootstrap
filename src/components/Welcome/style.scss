.welcome-splash {
  @include phone {
    position: fixed;
    top: 50%;
    transform: translate(0, -50%);
  }
  @include desktop {
    left: 50%;
    position: fixed;
    top: 50%;
    transform: translate(-50%, -50%);
  }

  .get-started {
    font-size: 18px;
    text-align: center;
  }

  .react-logo {
    width: 150px;

    @include phone {
      width: 75px;
    }
  }
}
