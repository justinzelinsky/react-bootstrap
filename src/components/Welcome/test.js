import '@testing-library/jest-dom/extend-expect';

import { fireEvent, render, screen } from '@testing-library/react';
import Welcome from 'components/Welcome';
import React from 'react';

test('Button click should increment count', function () {
  render(<Welcome />);

  expect(screen.getByRole('button')).toHaveTextContent('Or click me for fun! (0)');
  fireEvent.click(screen.getByRole('button'));
  expect(screen.getByRole('button')).toHaveTextContent('Or click me for fun! (1)');
});
