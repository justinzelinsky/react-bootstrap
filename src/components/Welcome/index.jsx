import './style.scss';

import ReactLogo from 'assets/images/react.svg';
import React from 'react';
import { Jumbotron } from 'react-bootstrap';

const Welcome = () => (
  <Jumbotron styleName="welcome-splash">
    <h1>
      <img alt="React" src={ReactLogo} styleName="react-logo" />
      React Starter
    </h1>

    <div styleName="get-started">
      Remove <code>components/Welcome</code> to begin working!
    </div>
  </Jumbotron>
);

export default Welcome;
