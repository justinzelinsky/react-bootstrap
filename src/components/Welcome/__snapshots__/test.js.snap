// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders 1`] = `
<div
  className="style-welcome-splash-1HWT5G jumbotron"
>
  <h1>
    <img
      alt="React"
      className="style-react-logo-3uUeWz"
      src="test-file-stub"
    />
    React Starter
  </h1>
  <div
    className="style-get-started-1byyDZ"
  >
    Remove 
    <code>
      components/Welcome
    </code>
     to begin working!
  </div>
</div>
`;
